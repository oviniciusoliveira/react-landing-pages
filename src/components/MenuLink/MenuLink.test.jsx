import { renderTheme } from '../../test/render-theme';
import { MenuLink } from '.';
import { screen } from '@testing-library/dom';

describe('<MenuLink />', () => {
  test('should render a link', () => {
    renderTheme(<MenuLink link="https://localhost">MenuLinkText</MenuLink>);
    const link = screen.getByRole('link', { name: 'MenuLinkText' });

    expect(link).toHaveAttribute('target', '_self');
    expect(link).toHaveAttribute('target', '_self');
  });

  test('should open link in a new tab', () => {
    renderTheme(
      <MenuLink link="https://localhost" newTab={true}>
        MenuLinkText
      </MenuLink>,
    );
    const link = screen.getByRole('link', { name: 'MenuLinkText' });

    expect(link).toHaveAttribute('target', '_blank');
  });

  test('should match snapshot', () => {
    renderTheme(<MenuLink link="https://localhost">MenuLinkText</MenuLink>);
    const link = screen.getByRole('link', { name: 'MenuLinkText' });

    expect(link).toMatchSnapshot();
  });
});
