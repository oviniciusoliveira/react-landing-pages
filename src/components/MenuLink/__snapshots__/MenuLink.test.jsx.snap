// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: block;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.78rem;
  height: 0.2rem;
  background: #dc143c;
  width: 0;
  left: 50%;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

<a
  class="c0"
  href="https://localhost"
  target="_self"
>
  MenuLinkText
</a>
`;
